gum/backend-linux/gumprocess-linux.c

```diff
@@ -440,6 +452,9 @@ gum_process_modify_thread (GumThreadId thread_id,
       }
     }
 
+    if (!(prev_dumpable < 0) && prev_dumpable != 1)
+      g_assert (!prctl (PR_SET_DUMPABLE, prev_dumpable));
```

Comment:
Code with important side-effects should never be inside assertions – which are actually disabled for release builds of Frida:

```suggestion
    {
      int res = prctl (PR_SET_DUMPABLE, prev_dumpable);
      g_assert (res == 0);
    }
```